pkg/util/kubernetes/apiserver/hpa_controller.go

```diff
@@ -39,34 +40,42 @@ type hpaToStoreGlobally struct {
 	data []custommetrics.ExternalMetricValue
 	m    sync.Mutex
 }
+
+// AutoscalersController is responsible for synchronizing objects from the Kubernetes
+// apiserver to build and cache the horizontal pod autoscalers in the cluster.
+// This allows to compute the External Metrics specified to autoscale pods and collect them from Datadog.
```

Comment:
`AutoscalersController is responsible for synchronizing horizontal pod autoscalers from the Kubernetes apiserver to determine the metrics that need to be provided by the custom metrics server. This controller also queries Datadog for the values of detected external metrics.`